### ISSUE TITLE
marker view anchoring incorrect after re-adding markers

### ISSUE THREAD

**Issue by cammace** [mapbox/mapbox-gl-native#6684]
Originally reported by a user, I was able to reproduce in the demo app. Steps to reproduce:

1. add markers to map
2. use `removeAnnotations()` to remove all the markers
3. re-add then again to the map and the anchoring will be off.

![ezgif com-video-to-gif](https://cloud.githubusercontent.com/assets/5652865/19323306/605d3776-908a-11e6-92cd-f4871dbff64a.gif)

I used the [`BasicMarkerViewActivity`](https://github.com/mapbox/mapbox-android-demo/blob/master/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/annotations/BasicMarkerViewActivity.java) with modified code:

```java
private boolean markers;

...

mapView.getMapAsync(new OnMapReadyCallback() {
      @Override
      public void onMapReady(final MapboxMap mapboxMap) {
        BasicMarkerViewActivity.this.mapboxMap = mapboxMap;
        addMarkers();
        markers = true;

        mapboxMap.setOnMapClickListener(new MapboxMap.OnMapClickListener() {
          @Override
          public void onMapClick(@NonNull LatLng latLng) {
            if (!markers) {
              addMarkers();
              markers = true;
            } else {
              mapboxMap.clear();
              markers = false;
            }
          }
        });

      }
    });
  }

private void addMarkers() {
    IconFactory iconFactory = IconFactory.getInstance(BasicMarkerViewActivity.this);
    Drawable iconDrawable = ContextCompat.getDrawable(BasicMarkerViewActivity.this, R.drawable.purple_marker);
    Icon icon = iconFactory.fromDrawable(iconDrawable);

    // The easiest way to add a marker view
    mapboxMap.addMarker(new MarkerViewOptions()
      .position(new LatLng(-37.821629, 144.978535)));

    // marker view using all the different options available
    mapboxMap.addMarker(new MarkerViewOptions()
      .position(new LatLng(-37.822829, 144.981842))
      .icon(icon)
      .rotation(90)
      .anchor(0.5f, 0.5f)
      .alpha(0.5f)
      .title("Hisense Arena")
      .snippet("Olympic Blvd, Melbourne VIC 3001, Australia")
      .infoWindowAnchor(0.5f, 0.5f)
      .flat(true));
  }
```

cc: @tobrun @zugaldia

**Comment by sanproh** [comment]
The issue is not reproduced on 4.1.0-beta.3 version of mapbox-android-sdk and initially appears in 4.1.0 release.

**Comment by tobrun** [comment]
@cammace what version of the sdk are you using? 
this seems a lot like a bug from previous release and I'm unable to reproduce:

![ezgif com-video-to-gif 68](https://cloud.githubusercontent.com/assets/2151639/19383797/7e758fd8-9206-11e6-99fe-2210d2e455b4.gif)

**Comment by cammace** [comment]
Was able to reproduce the bug in the testapp: 

![ezgif com-video-to-gif 4](https://cloud.githubusercontent.com/assets/5652865/19523728/67f4b8ba-95e9-11e6-87b2-432a1ce84b0c.gif)

Note that the GIF just ends quickly (markers not correcting itself at the end).

**Comment by tobrun** [comment]
@cammace I'm really not able to reproduce:

![ezgif com-video-to-gif 70](https://cloud.githubusercontent.com/assets/2151639/19555085/a43d3a0e-96bb-11e6-9b69-7e0d2902eddf.gif)

Could you debug [`MakerViewManager#Update()`](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/MarkerViewManager.java#L131)?

**Comment by cammace** [comment]
@tobrun has fixed the issue in #6809 : 

![ezgif com-video-to-gif 3](https://cloud.githubusercontent.com/assets/5652865/19692005/51fb0220-9aa4-11e6-86d4-e47d7efea24e.gif)

You'll see the fix once a new SNAPSHOT's built or when `beta.4`'s released.

**Comment by sanproh** [comment]
It works fine now on version 4.2.0-beta.4
Thanks a lot!
